pypsa_to_IPCC.py

```diff
@@ -121,136 +121,226 @@
+                 *n.links.p_nom_opt.filter(like ='CCGT').filter(like =country).filter(like=str(dict_var[v_type]))).sum())
+               var[v_type+'|Electricity|Gas'] += 0.001*((n.links.efficiency.filter(like ='gas CHP').filter(like =country).filter(like=str(dict_var[v_type]))
+                 *n.links.p_nom_opt.filter(like ='gas CHP').filter(like =country).filter(like=str(dict_var[v_type]))).sum())
+               var[v_type+'|Electricity|Gas|w/ CCS'] = 0.001*((n.links.efficiency.filter(like ='gas CHP CC').filter(like =country).filter(like=str(dict_var[v_type]))
+                 *n.links.p_nom_opt.filter(like ='gas CHP CC').filter(like =country).filter(like=str(dict_var[v_type]))).sum() )                                                 
+               var[v_type+'|Electricity|Gas|w/o CCS'] = (var[v_type+'|Electricity|Gas'] -
+                                         var[v_type+'|Electricity|Gas|w/ CCS']  )                                              
+                                                              
+               var[v_type+'|Electricity|Biomass'] = 0.001*((n.links.efficiency.filter(like ='solid biomass CHP').filter(like =country).filter(like=str(dict_var[v_type]))
+                 *n.links.p_nom_opt.filter(like ='solid biomass CHP').filter(like =country).filter(like=str(dict_var[v_type]))).sum())
+               var[v_type+'|Electricity|Biomass|w/ CCS']= 0.001*((n.links.efficiency.filter(like ='solid biomass CHP CC').filter(like =country).filter(like=str(dict_var[v_type]))
+                 *n.links.p_nom_opt.filter(like ='solid biomass CHP CC').filter(like =country).filter(like=str(dict_var[v_type]))).sum())   
+               var[v_type+'|Electricity|Biomass|w/o CCS'] = (var[v_type+'|Electricity|Biomass']    
+              -var[v_type+'|Electricity|Biomass|w/ CCS'])
+            
+               var[v_type+'|Electricity|Oil'] = 0.001*((n.links.efficiency.filter(like ='oil-').filter(like =country).filter(like=str(dict_var[v_type]))
+                 *n.links.p_nom_opt.filter(like ='oil-').filter(like =country).filter(like=str(dict_var[v_type]))).sum())
+            
+               """
+               Capacity : hydro (reservoir, ror)
+               """
+               #MW -> GW
+               var[v_type+'|Electricity|Hydro'] = 0.001*n.generators.p_nom_opt.filter(like ='ror').filter(like =country).filter(like=str(dict_var[v_type])).sum()
+               var[v_type+'|Electricity|Hydro'] += 0.001*n.storage_units.p_nom_opt.filter(like ='hydro').filter(like =country).filter(like=str(dict_var[v_type])).sum()
+               """
+               Capacity : storage (PHS, battery, H2 storage, thermal storage)
+               """
+               #MWh to GWh
+               var[v_type+'|Electricity|Storage|Pumped Hydro Storage'] = 0.001*(n.storage_units.p_nom_opt.filter(like ='PHS').filter(like =country).filter(like=str(dict_var[v_type])).sum())
```

Comment:
this reports the power capacity (instead of the intended energy capacity), you need to multiply by the attribute max_hours

the mentioned units are MWh to GWh, so this should be energy capacity, right?

Reply:
Okay, I had kept 1 variable with GW, but deleted it out so that all storage capacities have the GWh unit